packages/rs-drive-abci/src/abci/handlers.rs

```diff
@@ -528,48 +532,70 @@ where
         let _timer = crate::metrics::abci_request_duration("check_tx");
 
         let RequestCheckTx { tx, .. } = request;
-        let validation_result = self.platform.check_tx(tx.as_slice())?;
+        match self.platform.check_tx(tx.as_slice()) {
+            Ok(validation_result) => {
+                let validation_error = validation_result.errors.first();
+
+                let (code, info) = if let Some(validation_error) = validation_error {
+                    let serialized_error = platform_value!(validation_error
+                        .serialize()
+                        .map_err(|e| ResponseException::from(Error::Protocol(e)))?);
+
+                    let error_data = json!({
+                        "message": "Drive check_tx error",
```

Comment:
You don't need to pass the message. It will be recovered from the serialized error.